### ISSUE TITLE
Worker exited due to exception

### ISSUE THREAD

**Issue by BorzdeG** [jumperchen/zkidea#13]
IDEA: IU-144.3357.4

Worker exited due to exception
java.lang.NullPointerException
	at org.zkoss.zkidea.lang.ZulIconProvider.getIcon(ZulIconProvider.java:33)
	at com.intellij.util.IconUtil.getProvidersIcon(IconUtil.java:182)
	at com.intellij.util.IconUtil.access$200(IconUtil.java:49)
	at com.intellij.util.IconUtil$2.fun(IconUtil.java:145)
	at com.intellij.util.IconUtil$2.fun(IconUtil.java:136)
	at com.intellij.ui.IconDeferrerImpl.a(IconDeferrerImpl.java:87)
	at com.intellij.ui.IconDeferrerImpl.defer(IconDeferrerImpl.java:77)
	at com.intellij.util.IconUtil.getIcon(IconUtil.java:176)
	at com.intellij.psi.impl.ElementBase.getElementIcon(ElementBase.java:246)
	at com.intellij.psi.impl.ElementBase.computeIconNow(ElementBase.java:126)
	at com.intellij.psi.impl.ElementBase.access$000(ElementBase.java:51)
	at com.intellij.psi.impl.ElementBase$1.fun(ElementBase.java:62)
	at com.intellij.psi.impl.ElementBase$1.fun(ElementBase.java:55)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:261)
	at com.intellij.ui.DeferredIconImpl$1$1$1.run(DeferredIconImpl.java:135)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:122)
	at com.intellij.ui.DeferredIconImpl$1$1.run(DeferredIconImpl.java:132)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1185)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$2.run(ProgressIndicatorUtils.java:88)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$5.run(ProgressIndicatorUtils.java:134)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:130)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:95)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:85)
	at com.intellij.ui.DeferredIconImpl$1.run(DeferredIconImpl.java:129)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by jumperchen** [comment]
https://github.com/jumperchen/zkidea/blob/aa50b70e9e7b2f0702af3ea48c74e9db520b97a9/src/org/zkoss/zkidea/lang/ZulIconProvider.java#L33-L33

Strange for NPE ...

**Comment by BorzdeG** [comment]
parameter "file" as null?

**Comment by jumperchen** [comment]
if the file is null, it may be a bug in IntelliJ platform, anyway, we could prevent it just in case.

**Comment by BorzdeG** [comment]
fix:
old code: `file.getExtension().equalsIgnoreCase(ZulFileType.EXTENSION)`
new code: `ZulFileType.EXTENSION.equalsIgnoreCase(file.getExtension())`

Code clipping from 145.184:
```java
package com.intellij.openapi.vfs;
...
public abstract class VirtualFile ...
...
    @Nullable
    public String getExtension() {
        String name = this.getName();
        int index = name.lastIndexOf(46);
        return index < 0?null:name.substring(index + 1);
    }
```

**Comment by jumperchen** [comment]
ok

**Comment by jumperchen** [comment]
I released a 0.1.7 version, please give it a try - https://plugins.jetbrains.com/plugin/7855

**Comment by BorzdeG** [comment]
Yes, the error no longer exists. Thank you!
